### PR TITLE
Merge refactor

### PR DESCRIPTION
**Pull request by muhammadqamar**

### CODE REVIEW

File: src/components/Header/searchForm.js [hunk 1]
```diff
@@ -22,11 +23,18 @@ function SearchForm() {
...
             if (e.key === 'Enter') {
-              dispatcher(simpleSearchAction(simpleSearch, 0, 1000));
-              localStorage.setItem('loading', 'true');
-              if (simpleSearch) {
-                history.push('/search');
+              if(!simpleSearch.trim()){
+                Swal.fire("Search field is required")
+              }
+              else if(simpleSearch.length>255){
+                Swal.fire("character limit should be less then 255 ")
```
Comment (simonebelotti75) [inline src/components/Header/searchForm.js:30]: Please use Uppercase for better UX

File: src/components/Header/style.scss [hunk 2]
```diff
@@ -136,7 +136,10 @@
 }
 
 .tophd_left {
-  padding: 15px 0;
+ // padding: 15px 0;
+  img{
+    width:170px
```
Comment (simonebelotti75) [inline src/components/Header/style.scss:141]: Semi colon is missing

File: src/containers/Search/index.js [hunk 3]
```diff
@@ -158,7 +160,15 @@ function SearchInterface() {
 
                               <div
                                 onClick={() => {
-                                  if (searchInput) {
+                              
+                                  if(!searchInput.trim()){
+                                    Swal.fire("Search field is required")
+                                  }
+                                  else if(searchInput.length>255){
+                                    Swal.fire("character limit should be less then 255 ")
```
Comment (simonebelotti75) [inline src/containers/Search/index.js:168]: Uppercase

File: src/services/search.service.js [hunk 4]
```diff
@@ -9,7 +9,7 @@ const searchResult = (searchQuery, from, size) => httpService
   // .get(`https://refactored.curriki.org/api/api/v1/search/advanced?from=${from}&size=${size}&query=${searchQuery}`)
   .get(`/${apiVersion}/search/advanced?from=${from}&size=${size}&query=${searchQuery}`)
   .then(({ data }) => data)
-  .catch((err) => Promise.reject(err.response.data));
+  .catch((err) => err.response.data)
```
Comment (simonebelotti75) [inline src/services/search.service.js:12]: Don't we throw error?

**Review by simonebelotti75 — CHANGES_REQUESTED** [review]
There are also a lot of formatting issues

File: src/containers/Search/index.js (hunk 3 above)
Comment (muhammadqamar) [reply src/containers/Search/index.js:168]: fixed

File: src/components/Header/style.scss (hunk 2 above)
Comment (muhammadqamar) [reply src/components/Header/style.scss]: fixed

File: src/components/Header/searchForm.js (hunk 1 above)
Comment (muhammadqamar) [reply src/components/Header/searchForm.js]: fixed

File: src/services/search.service.js (hunk 4 above)
Comment (muhammadqamar) [reply src/services/search.service.js:12]: no we need a response here.  elastic validation

**Pull request merged**